doc/Manuals/ug-ss_x-road_6_security_server_user_guide.md

```diff
@@ -1646,47 +1704,81 @@ For example, to configure the parameters `archive-path` and `archive-max-filesiz
 
-#### 11.1.4 Archive encryption
 
-Archive files can be encrypted (`archive-encryption-enabled = true`) using GnuPG (OpenPGP compatible, RFC 4880). Please see e.g. [RFC 4880](https://www.ietf.org/rfc/rfc4880.txt) and [GnuPG](https://gnupg.org/) for more infomation about OpenPGP and GnuPG.
+#### 11.1.6 Archive Files
+
+Archive files (ZIP containers) are located in the directory specified by the configuration parameter `archive-path`. File names are in the format `mlog[-grouping]-X-Y-Z.zip[.gpg]`, where X is the timestamp (UTC time in the format `YYYYMMDDHHmmss`) of the first message log record, Y is the timestamp of the last message log record (records are processed in chronological order) and Z is 10 characters long alphanumeric random. If grouping is enabled, [-grouping] is a (possibly truncated and filename safe) member identifier. If encryption is enabled, the `[.gpg]` suffix is added.
+
+The most basic example of an archive file name when the encryption and grouping are switched off:
+
+    mlog-20210901100858-20210901100905-5wOUQnXK4G.zip
+
+When the archive encryption switched on:
+
+    mlog-20210901101923-20210901101926-pXAZkBjnAC.zip.gpg
+
+Switching on archive grouping by member produces the following:
+
+    mlog-INSTANCE_CLASS_CODE-20210901102251-20210901102254-yoA2z7EAkg.zip.gpg
+
+Finally, switching to archive grouping by subsystem gives:
+
+    mlog-INSTANCE_CLASS_CODE_CONSUMERSUBSYSTEM-20210901102521-20210901102524-m1s2v5qZRc.zip.gpg
+    mlog-INSTANCE_CLASS_CODE_PROVIDERSUBSYSTEM-20210901102521-20210901102524-7bZ0LoXNbu.zip.gpg
+
+
+#### 11.1.7 Archive Encryption and Grouping
+
+The archive files can be encrypted (`archive-encryption-enabled = true`) using GnuPG (OpenPGP compatible, RFC 4880). Please see e.g. [RFC 4880](https://www.ietf.org/rfc/rfc4880.txt) and [GnuPG](https://gnupg.org/) for more infomation about OpenPGP and GnuPG.
 
-When encryption is enabled, the archiving process expects to find a server signing key in `gpg-home-directory`. This key is used to sign the generated archive files. In case grouping is `none` or no per-member key is available the `archive-default-encryption-key` is used to encrypt archive files.
+By default, the produced archive files contain messages from all the security server's members, but it's possible to group the archives by member or by subsystem if needed. The grouping is controlled by the setting `archive-grouping`.
+
+When encryption is enabled, the archiving process expects to find a server signing key in `gpg-home-directory`. This key is used to sign the generated archive files and is automatically generated during the security server installation. In case grouping is `none` or no per-member key is available, the `archive-default-encryption-key` is used to encrypt archive files. When grouping is set to `member` or `subsystem`, the archiving process uses per-member keys if they are available in `archive-encryption-keys-dir`.
+
```

Comment:
I'd add a sub heading here, e.g.:

> **Generating per-member keys for archive encryption**

However, it doesn't have to be included in ToC, it could be regular text in bold.

Reply:
Added sub-heading in https://github.com/nordic-institute/X-Road/pull/1038/commits/65b6250e114c10c435a9196fb0e5b7029bd49653

Reply:
Key management documentation is partially outdated due to (conflicting) changes in XRDDEV-1767